group/lc/main.tf

```diff
@@ -3,13 +3,13 @@
 ## Creates security group
 resource "aws_security_group" "sg_asg" {
   description = "${var.stack_item_fullname} security group"
-  name_prefix = "${var.stack_item_label}-asg-"
+  name_prefix = "${var.stack_item_label}${var.sg_name_suffix}"
```

Comment:
Partially. It would look something like this.

`name_prefix="${length(var.lc_name_override) > 0 ? "" : \"${var.stack_item_label}-\"}"`
`name="${length(var.lc_name_override) > 0 ? var.lc_name_override : ""}"`